app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/views/DetailViewController.kt

```diff
@@ -0,0 +1,88 @@
+package au.com.gridstone.trainingkotlin.views
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import android.widget.ImageView
+import android.widget.ProgressBar
+import android.widget.TextView
+import androidx.core.os.bundleOf
+import androidx.core.view.isVisible
+import au.com.gridstone.trainingkotlin.BuildConfig.IMAGE_URL
+import au.com.gridstone.trainingkotlin.POKEMON
+import au.com.gridstone.trainingkotlin.R
+import au.com.gridstone.trainingkotlin.SCOPE_DETAIL
+import au.com.gridstone.trainingkotlin.data.Pokemon
+import au.com.gridstone.trainingkotlin.viewmodels.DetailViewModel
+import au.com.gridstone.trainingkotlin.viewmodels.DetailViewState
+import com.bluelinelabs.conductor.Controller
+import com.squareup.picasso.Picasso
+import kotlinx.coroutines.*
+import org.koin.core.component.*
+import org.koin.core.parameter.parametersOf
+import org.koin.core.qualifier.named
+import org.koin.core.scope.Scope
+import kotlin.coroutines.CoroutineContext
+
+class DetailViewController(bundle: Bundle) : Controller(bundle), KoinComponent,
+  CoroutineScope {
+  private val pokemon: Pokemon = bundle.getParcelable(POKEMON)!!
```

Comment:
avoid ‼️, instead be explicit about your assetion 
```kotlin
private val pokemon: Pokemon = requireNotNull(bundle.getParcelable(POKEMON)) { 
  "Details view controller attached without a pokemon" 
}
```

Reply:
yeah sure updating this to be explicit thanks.